src/System.Runtime/tests/System/HashCodeTests.netcoreapp.cs

```diff
@@ -0,0 +1,227 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Xunit;
+
+public static class HashCodeTests
+{
+#if SYSTEM_HASHCODE_TESTVECTORS
+    // These test vectors were created using https://asecuritysite.com/encryption/xxHash
+    // 1. Find the hash for "".
+    // 2. Find the hash for "abcd". ASCII "abcd" and bit convert to uint.
+    // 3. Find the hash for "abcd1234". ASCII [ "abcd", "1234"] and bit convert to 2 uints.
+    // n. Continue until "abcd0123efgh4567ijkl8901mnop2345qrst6789uvwx0123yzab".
+
+    // HashCode is not deterministic across AppDomains by design. This means that
+    // these tests can not be executed against the version that exists within
+    // CoreCLR. Copy HashCode and set m_seed to 0 in order to execute these tests.
+
+    [Theory]
+    [InlineData(0x02cc5d05U)]
+    [InlineData(0xa3643705U, 0x64636261U)]
+    [InlineData(0x4603e94cU, 0x64636261U, 0x33323130U)]
+    [InlineData(0xd8a1e80fU, 0x64636261U, 0x33323130U, 0x68676665U)]
+    [InlineData(0x4b62a7cfU, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U)]
+    [InlineData(0xc33a7641U, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U)]
+    [InlineData(0x1a794705U, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U)]
+    [InlineData(0x4d79177dU, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU)]
+    [InlineData(0x59d79205U, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U)]
+    [InlineData(0x49585aaeU, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U, 0x74737271U)]
+    [InlineData(0x2f005ff1U, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U, 0x74737271U, 0x39383736U)]
+    [InlineData(0x0ce339bdU, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U, 0x74737271U, 0x39383736U, 0x78777675U)]
+    [InlineData(0xb31bd2ffU, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U, 0x74737271U, 0x39383736U, 0x78777675U, 0x33323130U)]
+    [InlineData(0xa821efa3U, 0x64636261U, 0x33323130U, 0x68676665U, 0x37363534U, 0x6c6b6a69U, 0x31303938U, 0x706f6e6dU, 0x35343332U, 0x74737271U, 0x39383736U, 0x78777675U, 0x33323130U, 0x62617a79U)]
+    public static void HashCode_Add(uint expected, params uint[] vector)
+    {
+        var hc = new HashCode();
+        for (int i = 0; i < vector.Length; i++)
+            hc.Add(vector[i]);
+
+        Assert.Equal(expected, (uint)hc.ToHashCode());
+    }
+#endif
```

Comment:
It would be great to see a \#else which verifies that the seed is non-zero.  E.g. that the empty hash is not 0x02cc5d05.

If the seed is allowed to be randomly zero then it will have the potential for failure; but a 32-bit seed means a 1 in 4 billion chance of failure